lib/spack/spack/build_systems/python.py

```diff
@@ -211,149 +338,33 @@ def global_options(self, spec, prefix):
         """
         return []
 
-    def install(self, spec, prefix):
+    def install(self, pkg, spec, prefix):
         """Install everything from build directory."""
 
-        args = PythonPackage._std_args(self) + ["--prefix=" + prefix]
+        args = PythonPipBuilder.std_args(pkg) + ["--prefix=" + prefix]
 
-        for key, value in self.config_settings(spec, prefix).items():
+        for key, value in self.config_settings(pkg, spec, prefix).items():
             if spec["py-pip"].version < Version("22.1"):
                 raise SpecError(
                     "'{}' package uses 'config_settings' which is only supported by "
                     "pip 22.1+. Add the following line to the package to fix this:\n\n"
                     '    depends_on("py-pip@22.1:", type="build")'.format(spec.name)
                 )
+
             args.append("--config-settings={}={}".format(key, value))
-        for option in self.install_options(spec, prefix):
+
+        for option in self.install_options(pkg, spec, prefix):
             args.append("--install-option=" + option)
```

Comment:
For some reason, this isn't working, which is causing the `py-xgboost` build to fail in CI. Needs investigation.

Reply:
Should be fixed. Boils down to signatures being different  and needing special treatment, see https://github.com/spack/spack/pull/32068

Reply:
Thanks for the quick fix!